app/controllers/application_controller.rb

```diff
@@ -14,6 +14,16 @@ class ApplicationController < ActionController::Base
   # Helper methods that are also used in controllers
   helper_method :current_user, :logged_in?
 
+  def handle_unverified_request
+    raise ActionController::InvalidAuthenticityToken
+  end
+
+  rescue_from ActionController::InvalidAuthenticityToken do |exception|
+    reset_session
+    flash[:error] = "The csrf token for your session was wrong"
```

Comment:
This message is very developer-centric. The users of this system doesn't know what a csrf token is, so I would suggest having a simpler error message, like: "There was a problem processing your request. Please try again later.".